test/cases/schema-title-as-type-name.ts

```diff
@@ -0,0 +1,48 @@
+export var schema = {
+    "$schema": "http://json-schema.org/draft-03/schema",
+    "id": "http://mycompany.com/api/referencing.json",
+    "title": "Referencing",
+    "type": "object",
+    "properties": {
+        "ref": {
+            "$ref": "../../test/resources/ReferencedType.json"
```

Comment:
BTW, in the beginning I also thought that this could be tweaked with the `Settings.useFullReferencePathAsName` flag, but apparently this is not the case.